git/tag.go

```diff
@@ -2,12 +2,34 @@ package git
 
 import (
 	"fmt"
+	"io/ioutil"
 )
 
 // TagOptions is a stub for when more of Tag is implemented
 type TagOptions struct {
 	// Replace existing tags instead of erroring out.
 	Force bool
+
+	List bool
+}
+
+// List tags, if tagnames is specified only list tags which match one
+// of the patterns provided.
+func TagList(c *Client, patterns []string) ([]string, error) {
+	if len(patterns) != 0 {
+		return nil, fmt.Errorf("Tag list with patterns not implemented")
+	}
+
+	files, err := ioutil.ReadDir(c.GitDir.String() + "/refs/tags")
```

Comment:
I like to try to use filepath.Join() in these cases, just to be sure that there isn't "//" and the delimiter is the right one for windows.